app/script/view_model/content/InputBarViewModel.js

```diff
@@ -216,8 +214,29 @@ z.viewModel.content.InputBarViewModel = class InputBarViewModel {
     );
   }
 
-  addMention(userEntity) {
-    // TODO
+  addMention(userEntity, inputElement) {
+    const editedMention = this.editedMention();
+    const mentionData = {
+      length: userEntity.name().length + 1,
+      start: editedMention.start,
+      user_id: userEntity.id,
+    };
+    const mentionEntity = new z.message.MentionEntity(mentionData);
```

Comment:
Still a valid comment:

````
const mentionEntity = new z.message.MentionEntity().setUserIdMention(this.editedMention().start, userEntity.name().length + 1, userEntity.id);
````

Reply:
Then we have a problem with the API of the `MentionEntity`. Why is there 2 ways to feed data in there?

I would rather go for the constructor API (rather than a setter method) so that we can ensure that we build a correct `MentionEntity` from the start. WDYT?